docs/design/README.md

```diff
@@ -0,0 +1,276 @@
+# Alloy proposals
+
+## Introduction
+
+Grafana Alloy has a formalized procedure for proposing changes to Alloy.
+
+The proposal process is intended for user-facing features, enhancements, or any
+largely scoped internal change. Other changes typically do not need a proposal,
+such as minor code refactors.
+
+If a contributor is uncertain about whether a proposal is needed, it is
+preferable to create a proposal instead of jumping into a change that may be
+declined.
+
+## Process
+
+Proposals are the process of getting consensus on a problem and a solution to
+that problem. Proposals are the more detailed form of an enhancement request,
+which typically only include a problem statement but no suggested solution.
+
+Proposals typically include:
+
+* A clear problem statement.
+* A proposed solution to the problem statement.
+* If applicable, a list of alternative solutions with rationale for why the
+  alternative solutions were not selected.
+
+Proposals can still meet the above criteria while remaining brief. For example,
+this proposal would be sufficient:
+
+> I would like to use the batch processor component from OpenTelemetry
+> Collector in Grafana Alloy to reduce the number of outgoing network requests.
+> Please add the batch processor as a new Alloy component called
+> `otelcol.processor.batch`.
+
+Any existing issue can be turned into a proposal by adding a `proposal` label.
+
+At a high-level, proposals go through the following stages:
+
+1. **Issue**: A brief [issue][new-proposal] is created for the proposal. At
+   this stage, there is no need for a design document.
+
+2. **Discuss**: Public discussion on the proposal issue drives the proposal
+   towards one of three outcomes:
+
+   * Accept the proposal, or
+   * decline the proposal, or
+   * ask for a design document.
+
+   As public discussion occurs, the proposal is expected to be refined over
+   time to address incoming feedback.
+
+3. **Design document**: If requested, a design document is written, and the PR
+   for the design document becomes the new home for discussing the proposal.
+
+4. **Consensus**: Once comments and changes have slowed down, a final public
+   discussion aims to reach consensus for one of two outcomes:
+
+   * Accept the proposal, or
+   * decline the proposal.
+
+[new-proposal]: https://github.com/grafana/alloy/issues/new?assignees=&labels=proposal&projects=&template=proposal.yaml
+
+## Detail
+
+### Scope
+
+The proposal process is intended for user-facing features, enhancements or any
+largely scoped internal change. Other changes typically do not need a proposal,
+such as minor code refactors. If a contributor is uncertain about whether a
+proposal is needed, it is preferable to create a proposal instead of jumping
+into a change that may be declined.
```

Comment:
minor nit: replicated paragraph.

Reply:
Yeah, that was intentional; ideally most readers could stop reading before the detail section. Do you think I should rephrase this first instance so it's not literally a copy/paste of the same paragraph below?

Reply:
No, it's not a big deal, let's leave it as-is.